plugins/generic/usageStats/UsageStatsPlugin.inc.php

```diff
@@ -306,9 +306,44 @@ function getUsageEventCurrentDayLogName() {
 	 * @param $usageEvent array
 	 */
 	function _writeUsageEventInLogFile($usageEvent) {
+		$salt = null;
 		if ($this->getSetting(CONTEXT_ID_NONE, 'dataPrivacyOption')) {
-			// TODO: Get the salt
-			$salt = 'ToBeSolved';
+			// Salt management.
+			if (!Config::getVar('usageStats', 'salt_filepath')) return false;
+			$saltFilename = Config::getVar('usageStats', 'salt_filepath');
+			$currentDate = date("Ymd");
+			$saltFilenameLastModified = date("Ymd", filemtime($saltFilename));
+			$file = fopen($saltFilename, 'r');
+			$salt = trim(fread($file,filesize($saltFilename)));
+			fclose($file);
+			if (empty($salt) || ($currentDate != $saltFilenameLastModified)) {
+				if(function_exists('mcrypt_create_iv')) {
+					$newSalt = bin2hex(mcrypt_create_iv(16, MCRYPT_DEV_URANDOM|MCRYPT_RAND));
+				} elseif (function_exists('openssl_random_pseudo_bytes')){
+					$newSalt = bin2hex(openssl_random_pseudo_bytes(16, $cstrong));
+				} elseif (file_exists('/dev/urandom')){
+					$newSalt = bin2hex(file_get_contents('/dev/urandom', false, null, 0, 16));
+				} else {
+					$newSalt = mt_rand();
+				}
+				$file = fopen($saltFilename,'wb');
+				if (flock($file, LOCK_EX)) {
+					fwrite($file, $newSalt);
+					flock($file, LOCK_UN);
+				} else {
+					ScheduledTask::addExecutionLogEntry(__('plugins.generic.usageStats.saltWriteError', array()), SCHEDULED_TASK_MESSAGE_TYPE_ERROR);
```

Comment:
I don't think you need to pass the empty array when calling the translate method...